v0/destinations/delighted/transform.js

```diff
@@ -0,0 +1,225 @@
+const { EventType } = require("../../../constants");
+const {
+    getFieldValueFromMessage,
+    CustomError,
+    defaultRequestConfig,
+    extractCustomFields,
+    removeUndefinedAndNullValues,
+    getErrorRespEvents,
+    constructPayload,
+    getSuccessRespEvents,
+    getDestinationExternalID,
+    isEmptyObject
+} = require("../../util");
+
+const {
+    channelValidity,
+    eventValidity,
+    isValidEmail,
+    isValidPhone,
+    userValidity
+} = require("./util");
+const {
+    ENDPOINT,
+    DELIGHTED_EXCLUSION_FIELDS,
+    identifyMapping
+} = require("./config");
+
+const identifyResponseBuilder = async (message, {Config}) => {
+    const userId = getFieldValueFromMessage(message, "userIdOnly");
+    if (!userId) {
+      throw new CustomError(
+        "userId is required for identify",
+        400
+      );
+    }
+    let channel = getDestinationExternalID(message, "delightedChannelType") || Config.channel;
+    channel = channel.toLowerCase();
+
+    channel= channelValidity(channel,userId);
+    let payload = constructPayload(message, identifyMapping);
+
+    payload.send = false;
+    payload.channel = channel;
+    payload.delay = Config.delay || message.context.traits.delay || 0 ;
+    payload.last_sent_at = message.lastSentAt;
+
+    if(!payload.name){
+        const fName = getFieldValueFromMessage(message, "firstName" );
+        const lName = getFieldValueFromMessage(message, "lastName");
+        let name= fName.concat(lName);
```

Comment:
use `const` for variables which are not being reassigned.Also, lets try to maintain formatting.